sdk/core/core/src/az_http_header_validation_private.h

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file az_http_header_validation_private.h
+ *
+ * @brief This header defines a bit array that is used to validate whenever or not an ASCII char is
+ * valid within an http header name.
+ *
+ */
+
+#ifndef _az_HTTP_HEADER_VALIDATION_PRIVATE_H
+#define _az_HTTP_HEADER_VALIDATION_PRIVATE_H
+
+#include <az_span.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+// Bit array from ASCII to valid. Every zero in array means invalid character, while non-zeros
+// return the valid character.
+static const uint8_t az_http_valid_token[256] = {
+  0, // 0 ->  null
+  0, // 1 ->  start of heading
+  0, // 2 ->  start of text
+  0, // 3 ->  end of text
+  0, // 4 ->  end of transmission
+  0, // 5 ->  enquiry
+  0, // 6 ->  acknowledge
+  0, // 7 ->  bell
+  0, // 8 ->  backspace
+  0, // 9 ->  horizontal tab
+  0, // 10 ->  new line
+  0, // 11 ->  vertical tab
+  0, // 12 ->  new page
+  0, // 13 ->  carriage return
+  0, // 14 ->  shift out
+  0, // 15 ->  shift in
+  0, // 16 ->  data link escape
+  0, // 17 ->  device control 1
+  0, // 18 ->  device control 2
+  0, // 19 ->  device control 3
+  0, // 20 ->  device control 4
+  0, // 21 ->  negative acknowledge
+  0, // 22 ->  synchronous idle
+  0, // 23 ->  end of trans. block
+  0, // 24 ->  cancel
+  0, // 25 ->  end of medium
+  0, // 26 ->  substitute
+  0, // 27 ->  escape
+  0, // 28 ->  file separator
+  0, // 29 ->  group separator
+  0, // 30 ->  record separator
+  0, // 31 ->  unit separator
+  ' ', // 32 ->  space
```

Comment:
I don't think the space (' ') should be allowed within header names.

Reply:
header name can have spaces in the middle, like `this is my header:value`.  followinf RFC

Reply:
Can you point me to where it states that? I was looking at https://tools.ietf.org/html/rfc7230#section-3.2